### PR TITLE
fix authz on server listening on both ipv4 and ipv6 port

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
THis is a bug found by accessing hello world application from istio sample

### CODE REVIEW

File: bpf/kmesh/workload/sockops_tuple.c [hunk 1]
```diff
@@ -177,9 +177,6 @@ static inline bool conn_from_cni_sim_delete(struct bpf_sock_ops *skops)
 SEC("sockops")
 int record_tuple(struct bpf_sock_ops *skops)
 {
-	// only support IPV4
-	if (skops->family != AF_INET)
```
Comment (nlgwcy) [inline bpf/kmesh/workload/sockops_tuple.c:181]: 1. kmesh supports only IPv4.
2. In all-zero port listening scenario, socket on the server can process both IPv4 and IPv6 packets. When the server receives an IPv4 packet, the address is mapped to the IPv6 format (IPv4-mapped IPv6 address). However, the family may be AF_INET6. As a result, an error occurs.
Recommended solution:
1. IPv4-mapped IPv6 addresses
image
if (skops->family != AF_INET) -> if (skops->family != AF_INET || (IPV4-MAPPED_ADDR(skops->local_ip6))

Comment (hzxuzhonghu) [reply bpf/kmesh/workload/sockops_tuple.c:181]: good to know.

Can you check only skops->local_ip4?

Comment (hzxuzhonghu) [reply bpf/kmesh/workload/sockops_tuple.c:181]: @nlgwcy ITested, the 3rd u32 is 0xfffff0000

**Pull request merged**